source/simulator/initial_conditions.cc

```diff
@@ -308,10 +308,13 @@ namespace aspect
           fe_values.reinit (cell);
           const std::vector<Point<dim> > quadrature_points = fe_values.get_quadrature_points();
 
+          ComponentMask property_mask  (particle_property_manager->get_data_info().n_components(),false);
+          property_mask.set(particle_property,true);
+
           const std::vector<std::vector<double> > tracer_properties =
-            particle_interpolator->properties_at_points(*particles,quadrature_points,cell);
+            particle_interpolator->properties_at_points(*particles,quadrature_points,property_mask,cell);
```

Comment:
@gassmoeller - for clarification
If I was interested in implementing a new particle interpolation scheme, I can implement either of these two functions:

- `          virtual
          std::vector<std::vector<double> >
          properties_at_points(const std::multimap<types::LevelInd, Particle<dim> > &particles,
                               const std::vector<Point<dim> > &positions,
                               const typename parallel::distributed::Triangulation<dim>::active_cell_iterator &cell = typename parallel::distributed::Triangulation<dim>::active_cell_iterator()) const DEAL_II_DEPRECATED;`
- `          virtual
          std::vector<std::vector<double> >
          properties_at_points(const std::multimap<types::LevelInd, Particle<dim> > &particles,
                               const std::vector<Point<dim> > &positions,
                               const ComponentMask &selected_properties,
                               const typename parallel::distributed::Triangulation<dim>::active_cell_iterator &cell = typename parallel::distributed::Triangulation<dim>::active_cell_iterator()) const;
`

If I choose to implement the first function listed, then will the call
`particle_interpolator->properties_at_points(*particles,quadrature_points,property_mask,cell);`
be properly resolved? Is there some c++ magic happening here??